Mac plugin_lint_mac failing consistently on unrelated PR

On https://github.com/flutter/flutter/commit/0ce527eb91a37ef31b60ed6d4a9613310f2107d4, which doesn't look like it should have affected a plugin test. /cc @cyanglaz 

https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20plugin_lint_mac/5483/overview

```
[plugin_lint_mac] [STDOUT] stdout:         /opt/s/w/ir/cache/osx_sdk/XCode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target arm64-apple-macos10.11 -isysroot /opt/s/w/ir/cache/osx_sdk/XCode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -L/Users/chrome-bot/Library/Developer/Xcode/DerivedData/App-fnuhlnrabrbryscrhrnessyodoze/Build/Products/Release -L/opt/s/w/ir/cache/osx_sdk/XCode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -L/Users/chrome-bot/Library/Developer/Xcode/DerivedData/App-fnuhlnrabrbryscrhrnessyodoze/Build/Products/Release/test_plugin_swift -L/opt/s/w/ir/cache/osx_sdk/XCode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/usr/lib/swift -F/Users/chrome-bot/Library/Developer/Xcode/DerivedData/App-fnuhlnrabrbryscrhrnessyodoze/Build/Products/Release -F/opt/s/w/ir/x/t/CocoaPods-Lint-20220209-14900-7a9pgu-test_plugin_swift/Pods/FlutterMacOS -filelist /Users/chrome-bot/Library/Developer/Xcode/DerivedData/App-fnuhlnrabrbryscrhrnessyodoze/Build/Intermediates.noindex/App.build/Release/App.build/Objects-normal/arm64/App.LinkFileList -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @executable_path/../Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -Xlinker -rpath -Xlinker /opt/s/w/ir/cache/osx_sdk/XCode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -Xlinker -rpath -Xlinker @executable_path/../Frameworks -Xlinker -object_path_lto -Xlinker /Users/chrome-bot/Library/Developer/Xcode/DerivedData/App-fnuhlnrabrbryscrhrnessyodoze/Build/Intermediates.noindex/App.build/Release/App.build/Objects-normal/arm64/App_lto.o -fobjc-link-runtime -L/opt/s/w/ir/cache/osx_sdk/XCode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/chrome-bot/Library/Developer/Xcode/DerivedData/App-fnuhlnrabrbryscrhrnessyodoze/Build/Intermediates.noindex/App.build/Release/App.build/Objects-normal/arm64/App.swiftmodule -ObjC -ltest_plugin_swift -framework FlutterMacOS -framework Cocoa -lPods-App -Xlinker -dependency_info -Xlinker /Users/chrome-bot/Library/Developer/Xcode/DerivedData/App-fnuhlnrabrbryscrhrnessyodoze/Build/Intermediates.noindex/App.build/Release/App.build/Objects-normal/arm64/App_dependency_info.dat -o /Users/chrome-bot/Library/Developer/Xcode/DerivedData/App-fnuhlnrabrbryscrhrnessyodoze/Build/Intermediates.noindex/App.build/Release/App.build/Objects-normal/arm64/Binary/App
[plugin_lint_mac] [STDOUT] stdout:     ld: warning: ignoring file /opt/s/w/ir/x/t/CocoaPods-Lint-20220209-14900-7a9pgu-test_plugin_swift/Pods/FlutterMacOS/FlutterMacOS.framework/FlutterMacOS, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
[plugin_lint_mac] [STDOUT] stdout:     Undefined symbols for architecture arm64:
[plugin_lint_mac] [STDOUT] stdout:       "_FlutterMethodNotImplemented", referenced from:
[plugin_lint_mac] [STDOUT] stdout:           test_plugin_swift.TestPluginSwiftPlugin.handle(_: __C.FlutterMethodCall, result: (Any?) -> ()) -> () in libtest_plugin_swift.a(TestPluginSwiftPlugin.o)
[plugin_lint_mac] [STDOUT] stdout:           function signature specialization <Arg[1] = Dead, Arg[2] = Owned To Guaranteed> of function signature specialization <Arg[1] = [Closure Propagated : reabstraction thunk helper from @escaping @callee_unowned @convention(block) (@unowned Swift.Optional<Swift.AnyObject>) -> () to @escaping @callee_guaranteed (@in_guaranteed Swift.Optional<Any>) -> (), Argument Types : [@escaping @callee_unowned @convention(block) (@unowned Swift.AnyObject?) -> ()]> of test_plugin_swift.TestPluginSwiftPlugin.handle(_: __C.FlutterMethodCall, result: (Any?) -> ()) -> () in libtest_plugin_swift.a(TestPluginSwiftPlugin.o)
[plugin_lint_mac] [STDOUT] stdout:       "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
[plugin_lint_mac] [STDOUT] stdout:           objc-class-ref in libtest_plugin_swift.a(TestPluginSwiftPlugin.o)
[plugin_lint_mac] [STDOUT] stdout:     ld: symbol(s) not found for architecture arm64
[plugin_lint_mac] [STDOUT] stdout:     clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

/cc @godofredoc @yusuf-goog @jmagman @stuartmorgan

--- a/dev/devicelab/bin/tasks/plugin_lint_mac.dart
+++ b/dev/devicelab/bin/tasks/plugin_lint_mac.dart
@@ -239,7 +239,7 @@ Future<void> main() async {
       // Add the new plugins we just made.
       pubspecContent = pubspecContent.replaceFirst(
         '\ndependencies:\n',
-        '\ndependencies:\n  $objcPluginName:\n    path: $objcPluginPath\n  $swiftPluginName:\n    path: $swiftPluginPath\n  url_launcher:\n  url_launcher_macos:\n',
+        '\ndependencies:\n  $objcPluginName:\n    path: $objcPluginPath\n  $swiftPluginName:\n    path: $swiftPluginPath\n  url_launcher: 6.0.16\n  url_launcher_macos:\n',
       );
       objcPubspec.writeAsStringSync(pubspecContent, flush: true);
 

--- a/dev/devicelab/bin/tasks/plugin_lint_mac.dart
+++ b/dev/devicelab/bin/tasks/plugin_lint_mac.dart
@@ -399,7 +399,7 @@ Future<void> main() async {
 
       final File podfileLockFile = File(path.join(swiftAppPath, 'ios', 'Podfile.lock'));
       final String podfileLockOutput = podfileLockFile.readAsStringSync();
-      if (!podfileLockOutput.contains(':path: ".symlinks/plugins/url_launcher/ios"')
+      if (!podfileLockOutput.contains(':path: ".symlinks/plugins/url_launcher_ios/ios"')
         || !podfileLockOutput.contains(':path: Flutter')
           // test_plugin_objc no longer supports iOS, shouldn't be present.
         || podfileLockOutput.contains(':path: ".symlinks/plugins/test_plugin_objc/ios"')
@@ -417,7 +417,7 @@ Future<void> main() async {
 
       checkDirectoryExists(path.join(
         pluginSymlinks,
-        'url_launcher',
+        'url_launcher_ios',
         'ios',
       ));
 
@@ -447,7 +447,7 @@ void _validateIosPodfile(String appPath) {
 
   final File podfileLockFile = File(path.join(appPath, 'ios', 'Podfile.lock'));
   final String podfileLockOutput = podfileLockFile.readAsStringSync();
-  if (!podfileLockOutput.contains(':path: ".symlinks/plugins/url_launcher/ios"')
+  if (!podfileLockOutput.contains(':path: ".symlinks/plugins/url_launcher_ios/ios"')
     || !podfileLockOutput.contains(':path: Flutter')
     || !podfileLockOutput.contains(':path: ".symlinks/plugins/test_plugin_objc/ios"')
     || !podfileLockOutput.contains(':path: ".symlinks/plugins/test_plugin_swift/ios"')
@@ -479,7 +479,7 @@ void _validateIosPodfile(String appPath) {
 
   checkDirectoryExists(path.join(
     pluginSymlinks,
-    'url_launcher',
+    'url_launcher_ios',
     'ios',
   ));
 
@@ -516,7 +516,7 @@ void _validateMacOSPodfile(String appPath) {
   if (!podfileLockOutput.contains(':path: Flutter/ephemeral\n')
       || !podfileLockOutput.contains(':path: Flutter/ephemeral/.symlinks/plugins/url_launcher_macos/macos')
       || !podfileLockOutput.contains(':path: Flutter/ephemeral/.symlinks/plugins/test_plugin_swift/macos')
-      || podfileLockOutput.contains('url_launcher/')) {
+      || podfileLockOutput.contains('url_launcher_ios/')) {
     print(podfileLockOutput);
     throw TaskResult.failure('macOS Podfile.lock does not contain expected pods');
   }
@@ -546,7 +546,7 @@ void _validateMacOSPodfile(String appPath) {
 
   checkDirectoryNotExists(path.join(
     pluginSymlinks,
-    'url_launcher',
+    'url_launcher_ios',
   ));
 
   checkDirectoryExists(path.join(

